dotnet/samples/KernelSyntaxExamples/Program.cs

```diff
@@ -19,56 +20,85 @@ public static async Task Main()
+            { nameof(Example32_StreamingCompletion), async (_) => { await Example32_StreamingCompletion.RunAsync(); } },
+            { nameof(Example33_StreamingChat), async (_) => { await Example33_StreamingChat.RunAsync(); } },
+            { nameof(Example34_CustomChatModel), async (_) => { await Example34_CustomChatModel.RunAsync(); } },
+            { nameof(Example35_GrpcSkills), async (_) => { await Example35_GrpcSkills.RunAsync(); } },
+            { nameof(Example36_MultiCompletion), async (_) => { await Example36_MultiCompletion.RunAsync(); } },
+            { nameof(Example37_MultiStreamingCompletion), async (_) => { await Example37_MultiStreamingCompletion.RunAsync(); } },
+            { nameof(Example40_DIContainer), async (_) => { await Example40_DIContainer.RunAsync(); } },
+            { nameof(Example41_HttpClientUsage), async (_) => { await Example41_HttpClientUsage.RunAsync(); } },
+            { nameof(Example42_KernelBuilder), async (_) => { await Example42_KernelBuilder.RunAsync(); } },
+            { nameof(Example43_GetModelResult), async (_) => { await Example43_GetModelResult.RunAsync(); } },
+            { nameof(Example44_MultiChatCompletion), async (_) => { await Example44_MultiChatCompletion.RunAsync(); } },
+            { nameof(Example45_MultiStreamingChatCompletion), async (_) => { await Example45_MultiStreamingChatCompletion.RunAsync(); } },
+            { nameof(Example48_GroundednessChecks), async (_) => { await Example48_GroundednessChecks.RunAsync(); } },
+            { nameof(Example49_LogitBias), async (_) => { await Example49_LogitBias.RunAsync(); } },
+            { nameof(Example51_StepwisePlanner), async (_) => { await Example51_StepwisePlanner.RunAsync(); } },
+            { nameof(Example52_ApimAuth), async (_) => { await Example52_ApimAuth.RunAsync(); } },
+            { nameof(Example54_AzureChatCompletionWithData), async (_) => { await Example54_AzureChatCompletionWithData.RunAsync(); } },
+            { nameof(Example55_TextChunker), async (_) => { await Example55_TextChunker.RunAsync(); } },
+            { nameof(Example56_TemplateNativeFunctionsWithMultipleArguments), async (_) => { await Example56_TemplateNativeFunctionsWithMultipleArguments.RunAsync(); } },
+        };
+
+        // Filter and run examples
+        foreach (var example in examples)
+        {
+            if (string.IsNullOrEmpty(filter) || example.Key.Contains(filter, StringComparison.OrdinalIgnoreCase))
+            {
+                try
+                {
+                    Console.WriteLine($"Running {example.Key}...");
+                    await example.Value(cancellationToken).SafeWaitAsync(cancellationToken);
+                }
+                catch (ConfigurationNotFoundException ex)
+                {
+                    Console.WriteLine($"{ex.Message}. Skipping example {example.Key}.");
+                }
+            }
+        }
     }
```

Comment:
Using reflection you can massively simplify this implementation.

```suggestion
    private static async Task RunExamplesAsync(string? filteredExample, CancellationToken cancellationToken)
{
    var examplesToRun = (Assembly.GetExecutingAssembly().GetTypes())
        .Where(type => type.Name.StartsWith("Example", StringComparison.OrdinalIgnoreCase))
        .Select(type => type.Name).ToList();

      if (!string.IsNullOrWhiteSpace(filteredExample))
      {
          if (!examplesToRun.Contains(filteredExample))
          {
              Console.WriteLine($"Example \"{filteredExample}\" not found.");
              Console.WriteLine($"\nUse one of the available examples below:\n\n{string.Join('\n', examplesToRun)}");
              return;
          }
  
          // Clean up other examples
          examplesToRun.Clear();
  
          // Add selected one
          examplesToRun.Add(filteredExample);
      }
  
      foreach (var example in examplesToRun)
      {
          await RunSafeExampleAsync(example, cancellationToken);
      }
  }
  
  private static async Task RunSafeExampleAsync(string exampleClass, CancellationToken cancellationToken)
{
    var invokeExample = () =>
    {
        var method = Assembly.GetExecutingAssembly().GetType(exampleClass)?.GetMethod("RunAsync")
                     ?? throw new ArgumentOutOfRangeException(nameof(exampleClass), $"Example {exampleClass} not found");

        method.Invoke(null, null);
        return Task.CompletedTask;
    };

    try
    {
        Console.WriteLine($"Running {exampleClass}...");
        await SafeWaitAsync(invokeExample(), cancellationToken);
    }
    catch (ConfigurationNotFoundException ex)
    {
        Console.WriteLine($"{ex.Message}. Skipping example {exampleClass}.");
    }
}
```

Reply:
```
string? filter = args.Length > 0 ? args[0] : null;
```
Option 1: Change the `null` here to your default filter.

Option 2: Set Command-Line Arguments:

Right-click on your console application project in the Solution Explorer.
Choose "Properties" from the context menu.
In the project properties window, navigate to the "Debug" tab on the left.
Supply Command-Line Arguments:

In the "Command line arguments" field, enter the command-line arguments that your console application expects. Separate multiple arguments with spaces.

Reply:
Can you add these instructions to the README for this project?